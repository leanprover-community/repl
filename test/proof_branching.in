{"cmd": "theorem complex_and (p q r : Prop) (h1 : p ∧ q) (h2 : q → r) : p ∧ r := by sorry", "sorries": "individual"}

{"tactic": "apply And.intro", "proofState": 0}

{"tactic": "exact h1.left", "proofState": 1}

{"tactic": "apply h2", "proofState": 2}

{"tactic": "exact h1.right", "proofState": 3}

