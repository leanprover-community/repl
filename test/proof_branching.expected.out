{"sorries":
 [{"proofState": 0,
   "pos": {"line": 1, "column": 75},
   "parentDecl": "complex_and",
   "goals": ["p q r : Prop\nh1 : p ∧ q\nh2 : q → r\n⊢ p ∧ r"],
   "endPos": {"line": 1, "column": 80}}],
 "messages":
 [{"severity": "warning",
   "pos": {"line": 1, "column": 8},
   "endPos": {"line": 1, "column": 19},
   "data": "declaration uses 'sorry'"}],
 "env": 0}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 1,
 "goals":
 ["case left\np q r : Prop\nh1 : p ∧ q\nh2 : q → r\n⊢ p",
  "case right\np q r : Prop\nh1 : p ∧ q\nh2 : q → r\n⊢ r"]}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 2,
 "goals": ["case right\np q r : Prop\nh1 : p ∧ q\nh2 : q → r\n⊢ r"]}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 3,
 "goals": ["case right\np q r : Prop\nh1 : p ∧ q\nh2 : q → r\n⊢ q"]}

{"proofStatus": "Completed", "proofState": 4, "goals": []}

